### PR TITLE
Shuttle engines affecting travel time

### PR DESCRIPTION
**Pull request by AnturK**
Alright so this is initial version to gauge if this is even good idea. All details up to change.

Changes : 
- Engines are now movable - wrench + long welding time
- Losing all initial engines means 2x longer travel time
- Adding another set of initial engines means 2x faster travel time.
- Buying BYOS shuttle lets you buy one engine for 5000 points.

Should apply to all shuttles unless i missed something because the timing code for these is pretty bad and i feel like i'm compounding the bad code here.

Todo: 
~~- Reset engine counts on admin shuttle swap.~~ Actually not needed manipulator gets rid of mobile port anyway.
~~- Actually remove INDESTRUCTIBLE flag from engines - How much health should they have.~~ Done

### CODE REVIEW

**Comment by PKPenguin321** [comment]
They should have the same armor as airlocks and maybe 1.5-2x the health imo

**Comment by DoctoreMad** [comment]
Upgrading engines with Bluespace RPED should speed up travel time.
(Does this affect the cargo shuttle?)

**Comment by Bawhoppen** [comment]
shuttles were mine goddamnit, the pressure cooker sure is on now

**Comment by AnturK** [comment]
Yes, works with cargo shuttle and this was meant as backup to bomb-battery for upgrades.

WIP, because turns out i need to redo transit system.

**Comment by Cyberboss** [comment]
@AnturK

**Comment by AnturK** [comment]
Should be good now.

**Comment by KorPhaeron** [comment]
Can you build them?

**Comment by AnturK** [comment]
Right now no, only order from cargo if build your own shuttle is bought.

**Comment by KorPhaeron** [comment]
@AnturK 

You can probably farm unlimited of these by abusing the arrivals shuttle

**Comment by KorPhaeron** [comment]
Since it regenerates between trips

**Comment by AnturK** [comment]
Hmm, i'll fix this in a moment.

**Comment by AnturK** [comment]
You mixed something up @KorPhaeron , it just stops from moving when it's damaged it does not regen.

File: code/game/shuttle_engines.dm [hunk 1]
```diff
@@ -25,6 +99,10 @@
 /obj/structure/shuttle/engine/propulsion/burst
 	name = "burst"
 
+/obj/structure/shuttle/engine/propulsion/burst/cargo
+	state = ENGINE_UNWRENCHED
+	anchored = 0
```
Comment (lzimann) [inline code/game/shuttle_engines.dm]: `FALSE`

**Comment by KorPhaeron** [comment]
Whoops

I heard "docking for repairs" after I smashed a window on the shuttle and then left, I had assumed it was leaving again to get fixed

**Comment by AnturK** [comment]
It just docks permanently until it's pressure tight again (not actually pressurized so it's debatable if this is actually useful)

**Comment by Cyberboss** [comment]
Who's gonna repair it if not you? :honk:

**Pull request merged**